plugins/view-transitions/js/view-transitions.js

```diff
@@ -0,0 +1,186 @@
+/**
+ * Initializes view transitions for the current URL.
+ *
+ * @param {Object} config                       The view transitions configuration.
+ * @param {string} config.postSelector          General selector for post elements in the DOM.
+ * @param {Object} config.globalTransitionNames Map of selectors for global elements (queried relative to 'body')
+ *                                              and their view transition names.
+ * @param {Object} config.postTransitionNames   Map of selectors for post elements (queried relative to an element
+ *                                              identified by config.postSelector) and their view transition names.
+ */
+window.plvtInitViewTransitions = ( config ) => {
+	if ( ! window.navigation || ! ( 'CSSViewTransitionRule' in window ) ) {
```

Comment:
@swissspidy I get TypeScript warnings locally here, like `navigation` not being a valid property on `window`, etc. The same in regards to the various (valid) properties accessed on the `event` objects. Do you have any advice for how to fix them?

Reply:
Actually, looks like they show here too: https://github.com/WordPress/performance/actions/runs/14766566981/job/41459069785?pr=1999

Reply:
The Navigation API is still relatively new and only supported by Chromium-based browsers. TypeScript by default only ships such types if the feature is supported by at least 2 different browser engines.

Until that happens, you can use the community-maintained [`navigation-api-types`](https://www.npmjs.com/package/navigation-api-types) package.

Reply:
I tried that, but it looks like that package doesn't properly support the latest specification either. It doesn't support e.g. the `activation` property of https://developer.mozilla.org/en-US/docs/Web/API/Navigation, which is what's needed here. So using this package doesn't resolve the issue, for now I ended up manually typing it on a custom `ExtendedWindow` type.